test/tests/functional/pbs_multi_sched.py

```diff
@@ -1644,34 +1645,42 @@ def test_auto_sched_off_due_to_fds_limit(self):
         when number of open files per process are exhausted
         """
 
-        # get the number of open files per process
-        ret = self.du.run_cmd(
-            self.server.hostname, [
-                'ulimit', '-n'], as_script=True, logerr=False)
-        self.assertEqual(ret['rc'], 0)
-        open_files_limit = ret['out'][0]
+        if os.getuid() != 0 or sys.platform in ('cygwin', 'win32'):
+            self.skipTest("Test need to run as root")
 
-        # set the number of open files per process to 10
-        ret = self.du.run_cmd(
-            self.server.hostname, [
-                'ulimit', '-n', '10'], as_script=True, logerr=False)
-        self.assertEqual(ret['rc'], 0)
+        try:
+            # get the number of open files per process
+            (open_files_soft_limit, open_files_hard_limit) =\
+                resource.getrlimit(resource.RLIMIT_NOFILE)
+
+            # set the soft limit of number of open files per process to 10
+            resource.setrlimit(resource.RLIMIT_NOFILE,
+                               (10, open_files_hard_limit))
+        except (ValueError, resource.error):
+            self.assertFalse("Error in accessing system RLIMIT_ variales, "
+                             "test fails.")
```

Comment:
The correct call to this method is self.assertFalse(condition, msg).  While a message is always true, you really should do something like self.assertFalse(True, msg).

Reply:
Implemented.